### PR TITLE
allow cast use of broadcast intermediate

### LINKED ISSUE #1009: canSchedule returns false for legit kernel structure
## 🐛 Bug

`canSchedule` returns false on a legit kernel.

1. The issue is quite strange. `canSchedule` for normalization returns false for a kernel and later segmentation generates the same kernel with normalization scheduler. Since segmentation uses the same `canSchedule` function to validate segmented fusion, It looks like there's some graph mutation during segmentation that makes it acceptable for normalization scheduler. So we ended up recompiling when we have dynamic sizes.
2. I temporarily reverted our changes to `canSchedule` for normalization scheduler, the recompilation problem seems to go away. I pushed code changes to branch https://github.com/csarofeen/pytorch/tree/normalization_scheduler_revert_canSchedule

## To Reproduce

The script below should only generate a single set of kernels. However, the backward graph are recompiling kernel, instead of reusing it. it is recompiling the same kernel.

The fact that normalization scheduler `canSchedule()` returned false and segmenter later gives identical segmentation with normalization scheduler makes it obvious that there's a bug.

```
import torch

batch = 285
channels = 1024
dtype = torch.float16

class MyModule(torch.nn.Module):
  def __init__(self):
    super(MyModule, self).__init__()
    self.layer_norm = torch.nn.LayerNorm((channels,), device="cuda", dtype=dtype)

  def forward(self, x, y):
    o = x * 1.5 + y * 2
    o = self.layer_norm(o)
    return o

t = MyModule()
t_jit = torch.jit.script(t)

x = torch.randn(batch, channels, dtype=dtype, device="cuda").requires_grad_()
y = torch.randn(batch, channels, dtype=dtype, device="cuda").requires_grad_()

with torch.jit.fuser("fuser2"):
  for i in range(4):
    jit_o = t_jit(x, y)
    jit_o.sum().backward()

#print(t_jit.graph_for(x, y))

batch = 221

x = torch.randn(batch, channels, dtype=dtype, device="cuda").requires_grad_()
y = torch.randn(batch, channels, dtype=dtype, device="cuda").requires_grad_()

jit_o = t_jit(x, y)
jit_o.sum().backward()

torch.cuda.synchronize()
```

### PR DESCRIPTION
**Pull request by shmsong**
Fixes #1009

### CODE REVIEW

**Comment by csarofeen** [comment]
Closing in favor of: https://github.com/csarofeen/pytorch/pull/979
commit: https://github.com/csarofeen/pytorch/pull/979/commits/8d3ab30a3d35b1d70df76d8d0ff54d76b3b47b76
and
https://github.com/csarofeen/pytorch/pull/979/commits/2d57a7976876d12c6161512313ecfe21fa1d2c81
